unitTests/ui/ResponsiveComponents/ResponsiveTabsTest.ts

```diff
@@ -33,5 +49,120 @@ export function ResponsiveTabsTest() {
 
       expect(ResponsiveComponentsUtils.deactivateSmallTabs).toHaveBeenCalled();
     });
+
+    describe('when moving tabs to different section', () => {
+      const simulateOverflowing = () => {
+        Object.defineProperty(tabSection.el, 'clientWidth', {
+          value: 100,
+          writable: false
+        });
+
+        // This will simulate each tab being added progressively to the dropdown (25px each tab)
+        // reducing the element scroll width
+        let numberOfTimesCalled = -1;
+        Object.defineProperty(tabSection.el, 'scrollWidth', {
+          get: () => {
+            numberOfTimesCalled++;
+            return 125 - 25 * numberOfTimesCalled;
+          },
+          configurable: true
+        });
+      };
+
+      const simulateNotOverflowing = () => {
+        Object.defineProperty(tabSection.el, 'clientWidth', {
+          value: 100,
+          writable: false
+        });
+        Object.defineProperty(tabSection.el, 'scrollWidth', {
+          get: () => 50
+        });
+      };
+
+      const openTabContainer = () => {
+        $$(root.find('.coveo-dropdown-header')).trigger('click');
+        return $$(root.find('.coveo-tab-list-container'));
+      };
+
+      describe('when the tab section is overflowing', () => {
+        beforeEach(() => {
+          simulateOverflowing();
+          spyOn(root, 'width').and.returnValue(new ResponsiveComponents().getMediumScreenWidth() - 1);
+        });
+
+        it('it creates a dropdown header', () => {
+          responsiveTabs.handleResizeEvent();
+          expect(root.find('.coveo-dropdown-header')).not.toBeNull();
+        });
+
+        it('places tabs in the correct order inside and outside the dropdown', () => {
+          responsiveTabs.handleResizeEvent();
+          const container = openTabContainer();
+
+          const lastTabInDropdown = container.find(`div[data-id="4"]`);
+          expect(lastTabInDropdown).not.toBeNull();
+
+          range(0, 4).forEach(otherTabsNotInsideDropdown => {
```

Comment:
I would rename the var to use the singular since we are inside the `forEach`. Also, it looks like it is an id, so maybe: e.g. `tabNotInDropdownId` or `tabOutsideOfDropdownId`.